osf/models/node.py

```diff
@@ -1811,13 +1811,21 @@ def register_node(self, schema, auth, data, parent=None):
 
         for node_relation in original.node_relations.filter(child__is_deleted=False):
             node_contained = node_relation.child
+            if child_ids and node_contained._id not in child_ids:
+                children = node_contained.node_relations.filter(child__is_deleted=False, child__guids___id__in=child_ids)
```

Comment:
@sloria I couldn't really articulate this when we talked about it during 1-on-1, but the problem with validating within the register_node method is that you have to "look ahead". If we're validating if the node has a parent that isn't being registered and we skip those nodes, the checks there children preform never get run, so the truncated registration tree passes. This check fixes that, but it still requires some additional queries.